web/modules/custom/asset_distribution/asset_distribution.module

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * @file
+ * Contains asset_distribution.module.
+ */
+
+use \Drupal\Core\Entity\EntityInterface;
+use \Drupal\rdf_entity\RdfInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ *
+ * Handle the reverse reference of the solution-distribution entities.
+ */
+function asset_distribution_rdf_entity_presave(EntityInterface $entity) {
+  if ($entity->bundle() != 'asset_distribution') {
+    return;
+  }
+  // Handle reverse reference in the interoperability solution.
+  /** @var \Drupal\rdf_entity\RdfInterface $rdf_entity */
+  $rdf_entity = \Drupal::routeMatch()->getParameter('rdf_entity');
```

Comment:
Indeed. This could complicate things a bit. Added two more checks. One is regarding the route name. So this will be executed only when we are in the route of the adding a distribution through a solution page.
The other check is whether the entity is a new entity. This will ensure that this hook is not fired on entity update which would duplicate the value in the reference list.